sdk/identity/azure-identity/azure/identity/_internal/shared_token_cache.py

```diff
@@ -92,7 +92,7 @@ class SharedTokenCacheBase(ABC):
     def __init__(self, username=None, **kwargs):  # pylint:disable=unused-argument
         # type: (Optional[str], **Any) -> None
 
-        self._authority = kwargs.pop("authority", None) or KnownAuthorities.AZURE_PUBLIC_CLOUD
+        self._authority = kwargs.pop("authority", os.environ.get(EnvironmentVariables.AZURE_AUTHORITY_HOST)) or KnownAuthorities.AZURE_PUBLIC_CLOUD
```

Comment:
The impact here is tiny but FYI, function arguments are always evaluated. `os.environ.get` is called every time this line executes, whether `'authority' in kwargs` or not.

Reply:
I have updated the code according to your suggestion, there is no problem in the local test, but there are errors in CI.